params/config.go

```diff
@@ -358,24 +386,49 @@ type ChainConfig struct {
 	Ethash *EthashConfig `json:"ethash,omitempty"`
 	Clique *CliqueConfig `json:"clique,omitempty"`
 
-	// Scroll genesis extension: Use zktrie
-	Zktrie bool `json:"zktrie,omitempty"`
+	// Scroll genesis extension: enable scroll rollup-related traces & state transition
+	Scroll *ScrollConfig `json:"scroll,omitempty"`
+}
+
+type ScrollConfig struct {
+	// Use zktrie
+	UseZktrie bool `json:"useZktrie,omitempty"`
 
-	// Scroll genesis extension: Transaction fee vault address [optional]
+	// Maximum number of transactions per block [optional]
+	MaxTxPerBlock *int `json:"maxTxPerBlock,omitempty"`
+
+	// Transaction fee vault address [optional]
 	FeeVaultAddress *common.Address `json:"feeVaultAddress,omitempty"`
 
-	// Scroll genesis extension: enable EIP-2718 in tx pool.
+	// enable EIP-2718 in tx pool.
 	EnableEIP2718 bool `json:"enableEIP2718,omitempty"`
 
-	// Scroll genesis extension: enable EIP-1559 in tx pool, EnableEIP2718 should be true too.
+	// enable EIP-1559 in tx pool, EnableEIP2718 should be true too.
 	EnableEIP1559 bool `json:"enableEIP1559,omitempty"`
+}
 
-	// Scroll genesis extension: Maximum number of transactions per block [optional]
-	MaxTxPerBlock *int `json:"maxTxPerBlock,omitempty"`
+func (s *ScrollConfig) BaseFeeEnabled() bool {
+	if s == nil {
+		return false
```

Comment:
Not sure this will work well. But we probably shouldn't make too much effort supporting the `s == nil` case.

Reply:
resolved in https://github.com/scroll-tech/go-ethereum/pull/262/commits/17367e705f8c7ab3e855680aa48ed9a6a5fab322